ibm_was/datadog_checks/ibm_was/ibm_was.py

```diff
@@ -0,0 +1,103 @@
+# (C) Datadog, Inc. 2019
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import requests
+from lxml import etree
+
+from six import iteritems
+
+from datadog_checks.base import AgentCheck, is_affirmative
+from . import metrics, validation
+
+
+class IbmWasCheck(AgentCheck):
+
+    SERVICE_CHECK_CONNECT = "ibm_was.can_connect"
+
+    def check(self, instance):
+        validation.validate_config(instance)
+        collect_stats = self.setup_configured_stats(instance)
+        nested_tags, metric_categories = self.append_custom_queries(instance, collect_stats)
+
+        custom_tags = instance.get('custom_tags', [])
+
+        data = self.make_request("servers")
+        server_data_xml = etree.fromstring(data)
+        node_list = self.get_node_from_root(server_data_xml, "Node")
+
+        for node in node_list:
+            server_list = self.get_node_from_root(node, 'Server')
+            node_tags = list(custom_tags)
+            node_tags.append('node:{}'.format(node.get('name')))
+            for server in server_list:
+                server_tags = ['server:{}'.format(server.get('name'))]
+                server_tags.extend(node_tags)
+
+                for category, prefix in iteritems(metric_categories):
+                    self.log.debug("Collecting {} stats".format(category))
+                    if collect_stats.get(category):
+                        stats = self.get_node_from_name(server, category)
+                        self.process_stats(stats, prefix, metric_categories, nested_tags, server_tags)
+
+    def get_node_from_name(self, xml_data, path):
+        # XMLPath returns a list, but there should only be one element here since we start
+        # the search within a given Node/server
+        data = xml_data.xpath('//Stat[normalize-space(@name)="{}"]'.format(path))
+        if len(data):
+            return data[0]
+        else:
+            self.warning('Error finding {} stats in XML output.'.format(path))
+            return []
+
+    def get_node_from_root(self, xml_data, path):
+        return xml_data.xpath('//{}'.format(path))
+
+    # The XML will have Stat Nodes and Nodes that contain the metrics themselves
+    # We have to recursively go through each Stat Node to properly setup tags
```

Comment:
nit: can you write this in a neutral way, without "we"?

Reply:
Is `we` not neutral?

Reply:
I think you should describe what the code is doing, instead of saying "we" or "you"

Reply:
So, `This code recursively goes through each Stat Node to properly setup tags` ?

Reply:
I updated the comments to remove `we`